playout/libretime_playout/recorder.py

```diff
@@ -0,0 +1,402 @@
+import datetime
+import logging
+import math
+import os
+import re
+import signal
+import sys
+import time
+from datetime import timezone
+from pathlib import Path
+from queue import Queue
+from subprocess import PIPE, Popen
+from threading import Thread
+from typing import Any, Dict
+
+import mutagen
+import requests
+from libretime_api_client.v1 import ApiClient as LegacyClient
+
+from libretime_playout.config import PUSH_INTERVAL, RECORD_DIR, Config
+
+from .liquidsoap.client import LiquidsoapClient
+
+if sys.version_info < (3, 9):
+    from backports.zoneinfo import ZoneInfo
+else:
+    from zoneinfo import ZoneInfo
+
+logger = logging.getLogger(__name__)
+
+# TODO : add docstrings everywhere in this module
+
+
+def getDateTimeObj(time):
+    # TODO : clean up for this function later.
+    # - use tuples to parse result from split (instead of indices)
+    # - perhaps validate the input before doing dangerous casts?
+    # - rename this function to follow the standard convention
+    # - rename time to something else so that the module name does not get
+    #   shadowed
+    # - add docstring to document all behaviour of this function
+    timeinfo = time.split(" ")
+    date = [int(x) for x in timeinfo[0].split("-")]
+    my_time = [int(x) for x in timeinfo[1].split(":")]
+    return datetime.datetime(
+        date[0], date[1], date[2], my_time[0], my_time[1], my_time[2], 0, None
+    )
+
+
+class ShowRecorder(Thread):
+    name = "show_recorder"
+
+    def __init__(
+        self,
+        show_instance,
+        show_name,
+        filelength,
+        start_time,
+        config: Config,
+        legacy_client: LegacyClient,
+    ):
+        Thread.__init__(self)
+        self.legacy_client = legacy_client
+        self.config = config
+        self.filelength = filelength
+        self.start_time = start_time
+        self.show_instance = show_instance
+        self.show_name = show_name
+        self.p = None
+
+    def record_show(self):
+        length = str(self.filelength)
+        filename = self.start_time
+        filename = filename.replace(" ", "-")
+
+        joined_path = os.path.join(RECORD_DIR, filename)
+        filepath = f"{joined_path}.{self.config.playout.record_file_format}"
+
+        br = self.config.playout.record_bitrate
+        sr = self.config.playout.record_samplerate
+        c = self.config.playout.record_channels
+        ss = self.config.playout.record_sample_size
+
+        liq_client = LiquidsoapClient(
+            host=self.config.playout.liquidsoap_host,
+            port=self.config.playout.liquidsoap_port,
+        )
```

Comment:
Please use the existing liquidsoap client created in the main.

Reply:
Done